release notes/v0.31.0.md

```diff
@@ -0,0 +1,75 @@
+
+We are working on the proper documentation for this, as well as the overdue xk6 documentation about JS extensions, so keep a lookout on https://k6.io/docs/ .
+
+
+### Marking requests as failed ([#1856](https://github.com/loadimpact/k6/pull/1856))
+
+It's now possible to declare expected HTTP response statuses for either the entire test or for individual HTTP requests, and k6 will emit a new `http_req_failed` metric as well as tag HTTP metrics with `expected_response: <bool>`.
+
+For example:
+
+```javascript
+import http from 'k6/http';
+
+// Set expected statuses globally for all requests.
+http.setResponseCallback(http.expectedStatuses({min: 200, max: 399}, 418));
+
+export default function () {
+  // This request will be marked as failed.
+  http.get('https://httpbin.test.k6.io/status/400');
+  // This request will be considered as "passed" because of the responseCallback override.
+  http.get('https://httpbin.test.k6.io/status/400', { responseCallback: http.expectedStatuses(400) });
+}
+```
+
+Running this script will produce a summary like:
+
+```
+http_req_duration..............: avg=204.57ms min=203.31ms med=204.57ms max=205.82ms p(90)=205.57ms p(95)=205.7ms
+  { expected_response:true }...: avg=203.31ms min=203.31ms med=203.31ms max=203.31ms p(90)=203.31ms p(95)=203.31ms
+http_req_failed................: 50.00% ✓ 1   ✗ 1
+```
+
+Note the new `http_req_duration` sub-metric for passed requests only and the `http_req_failed` Rate metric.
+
+The `http.setResponseCallback()` is meant to allow arbitrary JS functions to process responses, but for now only the `http.expectedStatuses()` callback is supported.
+
+
+## Other enhancements and UX improvements
+
+- JS: Because of the awesome improvements to [goja](https://github.com/dop251/goja), the JS runtime k6 uses, it's no longer necessary for k6 to load [core.js](https://github.com/zloirock/core-js) to polyfill missing JS features when using the default [`--compatibility-mode=extended`](https://k6.io/docs/using-k6/javascript-compatibility-mode). So in v0.31.0 core.js has been dropped entirely, yielding some significant CPU and memory usage improvements. The actual numbers will depend on the use case, but users can expect [TODO: Mention a percentage range?] ([#1824](https://github.com/loadimpact/k6/pull/1824))
+- JS: Also because of goja improvements, some Babel plugins were disabled which should decrease CPU and memory usage even more. ([#1822](https://github.com/loadimpact/k6/pull/1822))
```

Comment:
I wouldn't say that there is any difference really .. .the things that we disabled were mostly:
1. async/generator stuff that were just not working
2. some not very heavy plugins which likely also had little effect.
I haven't benchmarked it and it is definitely faster, just I wouldn't argue it is a something that the end user will notice :shrug:

Reply:
I changed it to "minor performance benefits" in 039ef774, since there should be some and we should mention it.